internal/topology/check/compatibility_test.go

```diff
@@ -473,7 +473,7 @@ func TestClusterClassesAreCompatible(t *testing.T) {
 							builder.BootstrapTemplate(metav1.NamespaceDefault, "bootstrap1").Build()).
 						Build()).
 				Build(),
-			desired: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+			desired: builder.ClusterClass(metav1.NamespacePublic, "class1").
 				WithInfrastructureClusterTemplate(
 					builder.InfrastructureClusterTemplate(metav1.NamespaceDefault, "infra1").Build()).
```

Comment:
```suggestion
			desired: builder.ClusterClass("other"", "class1").
				WithInfrastructureClusterTemplate(
					builder.InfrastructureClusterTemplate("other"", "infra1").Build()).
```
I meant we have to change these refs. All referenced templates of a CC should be in the same Namespaces as the CC itself (same for the rest of this builder)